tests/unit/memefficiency.tcl

```diff
@@ -503,6 +503,99 @@ run_solo {defrag} {
             $rd_pubsub close
         }
 
+        test "Active defrag HFE: $type" {
+            r flushdb
+            r config resetstat
+            r config set hz 100
+            r config set activedefrag no
+            # TODO: Lower the threshold after defraging the ebuckets.
+            # Now just to ensure that the reference is updated correctly.
+            r config set active-defrag-threshold-lower 12
+            r config set active-defrag-cycle-min 65
+            r config set active-defrag-cycle-max 75
+            r config set active-defrag-ignore-bytes 1500kb
+            r config set maxmemory 0
+            r config set hash-max-listpack-value 512
+
+            # Populate memory with interleaving hash field of same size
+            set n 50000
+            set dummy_field "[string repeat x 400]"
+            set rd [redis_deferring_client]
+            for {set j 0} {$j < $n} {incr j} {
+                $rd hset h f$j $dummy_field
+                $rd hexpire h 9999999 1 f$j
+                $rd set k$j $dummy_field
+            }
+            assert_encoding hashtable h
+            for {set j 0} {$j < $n} {incr j} {
+                $rd read ; # Discard hset replies
+                $rd read ; # Discard hexpire replies
+                $rd read ; # Discard set replies
+            }
+
+            # Coverage for listpackex and eblist.
```

Comment:
probably I'm missing something but for eblist, don't you need to create a HT hash with volatile fields less than 16 items?

Reply:
i tried to get hashtable and listpack, eblist and ebrax to work in the same test, but i found it too complicated and intrusive, Let me think aganin.

Reply:
i limit the number of fields to 16 to make them in an eblist.
and all the hash keys are in a ebrax.

Reply:
okay, if you feel like this is not necessary, feel free to change it as you wish. I was just commenting as I saw the your comment for eblist.

Reply:
@tezc look like you missed the last commit, i made hash fields in the eblist and made all the hash keys in the ebrax, so we can cover both of them.
and make a small hash listpack to covert listpackex.

Reply:
okay, sounds great